SMatrixADT.c

```diff
@@ -130,15 +130,15 @@ int SMTX_read(SMatrixType A[])
             }
         }
     }
-    A->R[0]].value = n-1;
+    A->R[0].value = n-1;
```

Comment:
The "A->R[0]].value" statement contains an extra bracket, causing syntax errors.